packages/edit-site/src/components/page-templates/template-actions.js

```diff
@@ -36,34 +36,57 @@ export function useResetTemplateAction() {
 			isPrimary: true,
 			icon: backup,
 			isEligible: isTemplateRevertable,
-			async callback( template ) {
+			supportsBulk: true,
+			async callback( templates ) {
 				try {
-					await revertTemplate( template, { allowUndo: false } );
-					await saveEditedEntityRecord(
-						'postType',
-						template.type,
-						template.id
+					await Promise.all(
+						templates.map( ( template ) => {
```

Comment:
I think it's probably better to perform the actions for each template iteratively in one loop.
That means for every item(1. revert, 2. save) and then move on to the next item.

Additionally for bulk actions it's better to use [Promise.allSettled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled). Promise all will stop if on any rejection which means previous fulfilled promises were successful. So while for example we bulk delete 3 items and the third is rejected, the other two has been deleted.

Reply:
Feedback applied 👍